lib/active_merchant/billing/gateways/stripe_payment_intents.rb

```diff
@@ -423,17 +423,21 @@ def add_network_token_data(post_data, payment_method, options)
         post_data
       end
 
-      def add_network_token_cryptogram_and_eci(post, payment_method)
-        return unless adding_network_token_card_data?(payment_method)
+      def add_network_token_cryptogram_and_eci(post, payment_method, options)
+        return unless adding_network_token_card_data?(payment_method, options)
```

Comment:
This should be updated to `return unless payment_method.is_a?(NetworkTokenizationCreditCard) && options.dig(:stored_credential, :initiator) != 'merchant'`

Reply:
Only AP/GP transactions on the new flow would reach here since if it was on the old flow the [payment_method would be a token](https://github.com/activemerchant/active_merchant/blob/master/lib/active_merchant/billing/gateways/stripe_payment_intents.rb#L24)

Reply:
If we want to just be safe that it's only AP/GP new flow on this we could do something like

```
return unless payment_method.is_a?(NetworkTokenizationCreditCard) && options.dig(:stored_credential, :initiator) != 'merchant'
return if digital_wallet_payment_method?(payment_method) && options[:new_ap_gp_route] != true
```

Reply:
Great, done!